workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/metrics.py

```diff
@@ -46,7 +50,23 @@ def prepend_to_key(d, prefix):
 
 
 def weighted_mean(ds, w, dims):
-    return (ds * w).sum(dims) / w.sum(dims)
+    return (ds * w).sum(dims, skipna=True) / w.sum(dims, skipna=True)
+
+
+def _mask_array(
```

Comment:
I understand not wanting to couple too much between this and the `transform.py` module, but might be nice to not have this code duplicated. I'll leave it up to you. Could move the `mask_array` function to a new module and import it in both places.

Reply:
I'll keep this copy here, since I think the diags dataset that is used in `metrics.py` doesn't have a latitude data array. I guess it would still work if I moved the function to a shared location and passed some dummy arg for `latitude` when it's called here, but that could be confusing in the future if someone thinks they can use the `tropics` mask key for a metric.

Reply:
The dataset should have latitude (I think under `lat`) since I know it's used for making maps in the report.